decent-dao/fractal-interface
# Refactor DataLayer | DAO dashboard

## Description
Okay, I've got it. This PR was two different PRs but ended up changing tweaking alot of the same files, so went ahead and merged into one. 

There are quite a few different updates covered in this PR.
- Updates Dashboard components to use new Contest Provider.
- Tweaks `useFractalNode` so that state is updated by each async request when it is resolved versus waiting.
- Extracts Providers -> ProviderWrapper for seperating out  concerns.
- adds 'use client' in components dispatching or creating context. I'm not sure exactly what is going on here as they should already be considered 'client components' but adding these seems to speed up loading in these components
- Adds Redirect for invalidating DAOs
- Update DaoCreator components to work with new context provider
- removes `CacheProvider` and now toasts load with the proper UI? Why not sure?

But with this PR the app is at least viewing in a DAO, and is just a matter of updating the rest of the components to use the new setup!




## Notes
Known issue: back button or fractal logo redirectly to landing doesn't remove DAO from state. Will work to fix that in this or next PR (depending on timing of reviews)


## Issue / Notion doc (if applicable)



## Testing
Will have to pull down locally for visual testing, but should be able
- to load a dao from favorites
- create a new DAO either type
- connect/disconnect wallet


## Screenshots (if applicable)
![localhost_3000_(Desktop) (65)](https://user-images.githubusercontent.com/38386583/228647750-d63fd061-22b7-400d-9e47-62be1e009dda.png)
